WcaOnRails/app/controllers/roles_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class RolesController < ApplicationController
+  def role_list
+    respond_to do |format|
+      format.json do
+        user = User.find(params[:userId])
+        is_delegate = !user.delegate_status.nil?
+        active_roles = is_delegate ? [{'role': user.delegate_status}] : []
+        render json: {
+          activeRoles: active_roles,
+        }
+      end
+    end
+  end
+
+  def role_data
+    respond_to do |format|
+      format.json do
+        role_id = params[:roleId]
+        senior_delegates = User.where(delegate_status: "senior_delegate")
+        if role_id == 'new'
+          render json: {
+            roleData: {},
+            seniorDelegates: senior_delegates,
+          }
+        else
+          user = User.find(params[:userId])
+          render json: {
+            roleData: {
+              delegateStatus: user.delegate_status,
+              seniorDelegateId: user.senior_delegate.id,
+              location: user.location,
+            },
+            seniorDelegates: senior_delegates,
+          }
+        end
+      end
+    end
+  end
+
+  def role_update
+    respond_to do |format|
+      format.json do
+        user = User.find(params[:userId])
```

Comment:
If you do it like this:

```ruby
user_id = params.require(:userId)
user = User.find(user_id)
```

then Rails will automatically validate the request for you and throw a 403 if somebody doesn't send the `userId` param